packages/tools/devtools/devtools-view/src/components/SettingsView.tsx

```diff
@@ -62,9 +62,15 @@ const useStyles = makeStyles({
  */
 export function SettingsView(): React.ReactElement {
 	const { themeInfo, setTheme } = useThemeContext();
-
+	const telemetryConfig = new TelemetryConfigurationManager();
 	const styles = useStyles();
-	const [optedIn, setOptedIn] = useTelemetryOptIn();
+	const [optedIn, setOptedIn] = React.useState(telemetryConfig.isTelemetryOptedIn());
+
+	function handleTelemetryToggle(ev: React.ChangeEvent<HTMLInputElement>): void {
+		const checked = ev.target.checked;
+		setOptedIn(checked);
+		telemetryConfig.setTelemetryOptIn(checked);
+	}
```

Comment:
With the change we completely stopped using `useTelemetryOptIn` but it actually does something important that the new class doesn't: it updates the setting if someone manually changes the values in localStorage (lines 74-84 in TelemetryUtils.ts). I think we should stick to using that React hook.

Also, an issue I have with exporting the new class is that it lets things outside devtools-view also _change_ the state of telemetry opt-in, which they shouldn't be able to. I think we can simplify the devtools-view part of this change by just exporting a free function `export const isTelemetryOptInEnabled = () => getStorageValue(telemetryOptInKey);` from TelemetryUtils.ts, and using that for the check in the extension's logger. The rest of the changes in devtools-view can probably be reverted. What do you think?

Reply:
Yeah, that's a good point. The set function shouldn't be accessible outside of devtools-view. I like the idea of just exporting the optIn logic ie: `export const isTelemetryOptInEnabled = () => getStorageValue(telemetryOptInKey);`